scripts/neighbor_advertiser

```diff
@@ -82,16 +84,50 @@ def log_error(msg):
 config_db = None
 
 
-#
-# Get switch info and intf addr
-#
-
 def connect_config_db():
     global config_db
     config_db = ConfigDBConnector()
     config_db.connect()
 
 
+#
+# Check if a DIP returned from ferret is in any of this switch's VLANs
+#
+
+vlan_interface_query = None
+
+
+def is_dip_in_device_vlan(ferret_dip):
+    global vlan_interface_query
+
+    # Lazy load the vlan interfaces the first time we run this check.
+    if not vlan_interface_query:
+        vlan_interface_query = config_db.get_table('VLAN_INTERFACE')
+
+    ferret_dip = IPAddress(ferret_dip)
+
+    for vlan_interface in vlan_interface_query.iterkeys():
+        vlan_subnet = vlan_interface[1]
```

Comment:
As discussed, you can reuse this function for this check - `is_ip_prefix_in_key(self, key)`